monitoring/count-items-cronjob/alerts.yaml

```diff
@@ -0,0 +1,74 @@
+x-inputs:
+  - name: namespace
+    type: constant
+    value: zenko
+  - name: zenkoCRName
+    type: constant
+    value: artesca-data
+  - name: count_items_cronjob
+    type: constant
+    value: ops-count-items
+  - name: count_items_job_duration_threshold
+    type: config
+    value: 3600
+
+groups:
+  - name: count-items-cronjob/alerts.rules
+    rules:
+      - alert: CountItemsJobTakingTooLong
+        Expr: |
+          (
+            time() - 
+            label_replace(
+              max(
+                kube_job_status_start_time
+                * ON(job_name,namespace) GROUP_RIGHT()
+                kube_job_owner{owner_name="${zenkoCRName}-${count_items_cronjob}"}
+              )
+              BY (job_name, owner_name, namespace)
+              == ON(owner_name) GROUP_LEFT()
+              max(
+                kube_job_status_start_time
+                * ON(job_name,namespace) GROUP_RIGHT()
+                kube_job_owner{owner_name="${zenkoCRName}-${count_items_cronjob}"}
+              )
+              BY (owner_name),
+              "cronjob", "$1", "owner_name", "(.+)")
+          ) > ${count_items_job_duration_threshold}
+        Labels:
+          severity: warning
+        Annotations:
+          description: |
+            Job {{$labels.job_name}} is taking more than 1h to complete.
+            This means that consumption information will be delayed
+            and the reported metrics might not be accurate anymore.
+          summary: "Job {{$labels.job_name}} didn't finish after 1h"
+
+      - alert: CountItemsJobFailed
+        Expr: |    
+          clamp_max (
+            label_replace(
+              max(
+                kube_job_status_start_time
+                * ON(job_name,namespace) GROUP_RIGHT()
+                kube_job_owner{owner_name="${zenkoCRName}-${count_items_cronjob}"}
+              )
+              BY (job_name, owner_name, namespace)
+              == ON(owner_name) GROUP_LEFT()
+              max(
+                kube_job_status_start_time
+                * ON(job_name,namespace) GROUP_RIGHT()
+                kube_job_owner{owner_name="${zenkoCRName}-${count_items_cronjob}"}
+              )
+              BY (owner_name),
+              "cronjob", "$1", "owner_name", "(.+)")
+            , 1)
+          * ON(job_name) GROUP_LEFT()
+          (kube_job_status_failed > 0)
```

Comment:
This can be simplified using the `kube_job_status_failed` metric, it's basically a gauge that gives you the number of failed jobs. You can select the job using regex, it should look something like `kube_job_status_failed{job_name=~"artesca-data-ops-count-items-.*"}`.

One thing to keep in mind is that we already have alerts for failed jobs, so you might want to check if this alert is needed.

Reply:
cool, thanks for letting me know! seems the fail check not needed anymore and I enabled `removeFailedJobsOnSuccess` on those cronjobs, it can be alerted by the default rule already